### ISSUE TITLE
Energy modification checks and consolidation

### ISSUE THREAD

**Issue by kyrsjo** [SixTrack/SixTrack#128]
Currently there are two ways of modifying the reference energy of the machine in SixTrack - the energy ramping controlled by `nde(1)` and `nde(2)`, as well as also used in `nwr(2)` and `nwr(3)`, possibly `POST` with `icow`=1 -- and the new mechanism using DYNK and introduced in #110 .

It would be good to (1) cross-check that these really give the right result (to do this one should start tracking at cavity), and (2) if they do, get rid of the thin6dua and thick6dua code. There should probably be either a conversion script or the presence of these parameters in the input should create 

Another thing that should probably be better tested is how the BEAM element is affected by the ramping since this element does quite a bit of initialisation, and one could probably also assume that both the weak and the strong beam is ramped together, squeezing the strong beam as well as the weak... This would couple to #11 , including a DYNK-ifying of that.

**Comment by kyrsjo** [comment]
Bug: Please update yv as well as energy variables.

Example from thin6d, accelerating cavity (label 30):
```
            yv(1,j)=(ejf0v(j)/ejfv(j))*yv(1,j)                           !hr01
            yv(2,j)=(ejf0v(j)/ejfv(j))*yv(2,j)                           !hr01
```

**Comment by Pingul** [comment]
I am not sure if I am thinking this wrong or not, but from what I can see I do not change the momentum of the particles in DYNK (as the momentum is an absolute value and not related to the reference energy). Currently it looks like

```
            do j = 1, napx
              ! updating values
              dpsv(j) = (ejfv(j) - e0f)/e0f
              dpsv1(j) = (dpsv(j)*c1e3)/(one + dpsv(j))
              dpd(j) = one + dpsv(j)
              dpsq(j) = sqrt(dpd(j))
              oidpsv(j) = one/(one + dpsv(j))
              rvv(j) = (ejv(j)*e0f)/(e0*ejfv(j))
            enddo
```

Adding the lines above should thus not have an effect. Am I missing something?

**Comment by kyrsjo** [comment]
True, agreed. Then there is no bug. We still need to understand the "jumping back and forth" in z tough. That is really strange...

**Comment by rdemaria** [comment]
I miss what you mean by "jumping back and forth" in z

**Comment by Pingul** [comment]
There was a problem where the z-coordinate of a particle would "jump back and forth" when DUMPing every turn. We thought this was an issue related to the DYNK ramping, but it turns out that it was due to an error in the fort.3 -- the length of LHC was 20 cm too short. Fixing that issue, solved the problem.

You could probably close this issue.

**Comment by kyrsjo** [comment]
Argh. #151 strikes again...

**Comment by rdemaria** [comment]
shall we close this one then?

**Comment by Pingul** [comment]
I think the bug @kyrsjo mentioned is solved, but I can't speak for the issue itself. The DYNK ramping seems to produce results that are in alignment with what expected, but this has not been verified in a systematic setting. I can't really comment on the other issues raised in the OP.

**Comment by rdemaria** [comment]
I then close it...

**Comment by rdemaria** [comment]
Being address in #472

**Comment by rdemaria** [comment]
Merged in #472

**Comment by JamesMolson** [comment]
Reviewing some of the array compaction code, I realized that we should maybe also call synutchk not just envarsv. See code `thick6d` just above `call dump_lines`. /// Kyrre
